### PR TITLE
Added BindingOperations.DoNothing

### LINKED ISSUE #1287: Binding.DoNothing is missing
Binding enum to multiple switches is simpler with DoNothing.

One could change BindingExpression to add such behavior. For example:

`if (converted == Binding.DoNothing)
                    {
                          return;
                    }`

### PR DESCRIPTION
**Pull request by jp2masa**
I'm opening this PR because I need this feature, but I'm not totally sure it matches the expected behavior. I read WPF docs and tried to match that behavior, but it's not totally clear and also there may be a better option for some cases, so all feedback is welcome of course!

## What does the pull request do?

Adds `BindingOperations.DoNothing`, and also tries to fix `AvaloniaProperty.UnsetValue` behavior on `MultiBinding`s (if I understand WPF docs correctly, it should always fallback to `FallbackValue`, if it exists, i.e. the property exists, not value != null).

## What is the current behavior?

There's no way to cancel current binding update.

## What is the updated/expected behavior with this PR?

`Binding.DoNothing` allows to cancel binding updates, on bindings and converted values on bindings.

## Checklist

- [x] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Avaloniaui.net with user documentation

## Breaking changes

Maybe breaks some existing binding behavior.

## Fixed issues
Fixes #1287

### CODE REVIEW

File: tests/Avalonia.Markup.Xaml.UnitTests/Data/BindingTests.cs [hunk 1]
```diff
@@ -34,10 +24,16 @@ public void Binding_With_Null_Path_Works()
                 var window = (Window)loader.Load(xaml);
                 var textBlock = window.FindControl<TextBlock>("textBlock");
 
-                window.DataContext = "foo";
                 window.ApplyTemplate();
 
+                window.DataContext = "foo";
                 Assert.Equal("foo", textBlock.Text);
```
Comment (grokys) [inline tests/Avalonia.Markup.Xaml.UnitTests/Data/BindingTests.cs]: I don't think this is really the right place for this test - this test is testing that a `{Binding}` without a path works, which has nothing to do with `BindingOperations.DoNothing`. Also setting `DataContext` to `BindingOperations.DoNothing` seems a rather strange test...

**Review by grokys — CHANGES_REQUESTED** [review]
This is great, thanks @jp2masa. Just one problem with the unit tests and it'll be good to merge I think.

File: tests/Avalonia.Markup.Xaml.UnitTests/Data/BindingTests.cs (hunk 1 above)
Comment (jp2masa) [reply tests/Avalonia.Markup.Xaml.UnitTests/Data/BindingTests.cs]: I totally understand that, and I agree it seems strange. When I wrote this, the idea was to test some more scenarios with bindings with null path, although thinking better it really doesn't belong here.

The main purpose for this is to define/illustrate the behavior of `DoNothing` with this PR, which in some cases may not seem to make sense, but they're cases which have to be handled (i.e. setting `DataContext` or a property in the `DataContext` to `DoNothing`).

I'll move this test elsewhere, but first I'd like to ask if you believe this behavior looks correct? `DoNothing` is a really special value, and I think setting a property in the view model to this value wouldn't make sense, but if someone does it, they will get the expected behavior I believe.

Comment (grokys) [reply tests/Avalonia.Markup.Xaml.UnitTests/Data/BindingTests.cs]: Yeah I think setting a property on a view model to `DoNothing` would be a decent thing to unit test.

**Review by grokys — APPROVED** [review]

**Pull request merged**